### DISCUSSION TITLE
wf_event_focus_in(wfc) called twice?

### DISCUSSION THREAD

**Discussion by garbb** [FreeRDP/FreeRDP#8268] [Q&A]
@erbth
https://github.com/FreeRDP/FreeRDP/blob/479e891545473f01c187daffdfa05fc752b54b72/client/Windows/wf_client.c#L1021-L1025

Does anyone know why this is called twice here? Is this just a mistake?

**Comment by erbth** [comment]
I don't remember exactly, but I think I copied the code from the X11 client, which did (and still does) call its *_focus_in function twice (see https://github.com/FreeRDP/FreeRDP/blob/master/client/X11/xf_client.c#L1537). A comment in the X11 client states that sending the keyboard indicators sync event twice helped with certain Windows versions (when connecting to Windows) - probably the keyboard indicators had not been synchronized reliably when sending the event only once.

I probably should have added a comment here in wfreerdp as well - sorry for that.

**Reply by garbb** [reply]
wow, I did not expect you to remember something from that long ago. Thanks.
